Just tried linking the new version abseil-cpp, instead of fetching and compiling it in place. But it brokes.

Full build log:

```
this derivation will be built:
  /nix/store/m97k34xv7cfmsp09icx7jcdv158a0xrw-fcitx5-mozc-2.26.4220.102.drv
building '/nix/store/m97k34xv7cfmsp09icx7jcdv158a0xrw-fcitx5-mozc-2.26.4220.102.drv'...
unpacking sources
unpacking source archive /nix/store/50rjw6qvd56x4qhhh1m88l92pak194kq-source
source root is source
Archive:  /nix/store/dgvgwargipiam5wk811f1s71521f9bza-x-ken-all-202011.zip
  inflating: source/src/x-ken-all.csv  
Archive:  /nix/store/pnm9ky4942qym52wmb85a683nd9vwb31-jigyosyo-202011.zip
  inflating: source/src/JIGYOSYO.CSV  
setting SOURCE_DATE_EPOCH to timestamp 1610952992 of file source/src/win32/tip/tip_ui_renderer_immersive.h
patching sources
applying patch /nix/store/38krcd7x93d59gk1sylyd5i6qf8dikfh-0007-Update-src-base-absl.gyp.patch
patching file src/base/absl.gyp
patching file src/base/base.gyp
patching file src/base/base_test.gyp
patching file src/composer/composer.gyp
patching file src/composer/composer_test.gyp
patching file src/converter/converter.gyp
patching file src/converter/converter_test.gyp
patching file src/data_manager/data_manager.gypi
patching file src/data_manager/data_manager_base.gyp
patching file src/dictionary/dictionary.gyp
patching file src/dictionary/dictionary_base.gyp
patching file src/dictionary/dictionary_test.gyp
patching file src/dictionary/file/dictionary_file.gyp
patching file src/engine/engine.gyp
patching file src/gui/gui.gyp
patching file src/prediction/prediction.gyp
patching file src/rewriter/calculator/calculator.gyp
patching file src/rewriter/rewriter.gyp
patching file src/rewriter/rewriter_test.gyp
patching file src/session/session.gyp
patching file src/storage/louds/louds.gyp
patching file src/storage/louds/louds_test.gyp
patching file src/storage/storage.gyp
patching file src/testing/testing.gyp
patching file src/unix/ibus/ibus.gyp
patching file src/win32/base/win32_base.gyp
configuring
INFO: Generating version definition file...
INFO: Version string is 2.26.4220.102
INFO: Running: pkg-config --exists ibus-1.0 >= 1.4.1
INFO: pkg-config --exists ibus-1.0 >= 1.4.1 failed
INFO: removing ibus.gyp.
INFO: Building GYP command line...
INFO: Running: pkg-config --exists Qt5Core Qt5Gui Qt5Widgets
INFO: Running GYP...
INFO: Running: /nix/store/9i978jmxclf17m2gr7w0sx7ybbvkgmsp-python3.8-gyp-2020-05-12/bin/gyp --depth=. --include=./gyp/common.gypi -D abs_depth=/build/source/src -D ext_third_party_dir=/build/source/src/third_party -D python=/nix/store/2wwb85hrvvfzkhrlwcj3pqi60nlbi9cq-python3-3.8.7/bin/python ./base/absl.gyp ./base/base.gyp ./base/base_test.gyp ./client/client.gyp ./client/client_test.gyp ./composer/composer.gyp ./composer/composer_test.gyp ./config/config.gyp ./config/config_test.gyp ./converter/converter.gyp ./converter/converter_base.gyp ./converter/converter_main.gyp ./converter/converter_test.gyp ./data/test/session/scenario/scenario.gyp ./data/test/session/scenario/usage_stats/usage_stats.gyp ./data_manager/data_manager.gyp ./data_manager/data_manager_base.gyp ./data_manager/data_manager_test.gyp ./data_manager/oss/oss_data_manager.gyp ./data_manager/oss/oss_data_manager_base.gyp ./data_manager/oss/oss_data_manager_test.gyp ./data_manager/testing/mock_data_manager.gyp ./data_manager/testing/mock_data_manager_base.gyp ./data_manager/testing/mock_data_manager_test.gyp ./dictionary/dictionary.gyp ./dictionary/dictionary_base.gyp ./dictionary/dictionary_test.gyp ./dictionary/file/dictionary_file.gyp ./dictionary/file/dictionary_file_test.gyp ./dictionary/system/system_dictionary.gyp ./dictionary/system/system_dictionary_test.gyp ./engine/engine.gyp ./engine/engine_test.gyp ./gui/gui.gyp ./gyp/tests.gyp ./ios/ios.gyp ./ipc/ipc.gyp ./mac/mac.gyp ./net/jsoncpp.gyp ./net/net.gyp ./net/net_test.gyp ./prediction/prediction.gyp ./prediction/prediction_base.gyp ./prediction/prediction_test.gyp ./protobuf/protobuf.gyp ./protocol/protocol.gyp ./renderer/renderer.gyp ./request/request.gyp ./rewriter/calculator/calculator.gyp ./rewriter/rewriter.gyp ./rewriter/rewriter_base.gyp ./rewriter/rewriter_test.gyp ./server/server.gyp ./session/session.gyp ./session/session_base.gyp ./session/session_test.gyp ./storage/storage.gyp ./storage/storage_test.gyp ./testing/testing.gyp ./transliteration/transliteration.gyp ./transliteration/transliteration_test.gyp ./unix/emacs/emacs.gyp ./unix/fcitx5/fcitx5.gyp ./usage_stats/usage_stats_base.gyp ./usage_stats/usage_stats_test.gyp -D branding=Mozc -D use_qt=YES -D qt_dir= -D use_wix=NO -D build_base=/build/source/src/out_linux -D build_short_base=out_linux -D warn_as_error=0 -D channel_dev=1 -D target_platform=Linux -D server_dir=/nix/store/m0ylhv861rxpfikx46m64iw6apv0hryh-fcitx5-mozc-2.26.4220.102/lib/mozc --generator-output=. -G output_dir=out_linux
INFO: Done
building
INFO: Running: ninja -C out_linux/Release mozc_server
ninja: Entering directory `out_linux/Release'
[381/524] LINK gen_usage_rewriter_dictionary_mainified_sha1.oer_bitarray_main.onoun_prefix_data.ob6ffbdbdc511610964
FAILED: gen_usage_rewriter_dictionary_main 
clang++ -pthread -o gen_usage_rewriter_dictionary_main -Wl,--start-group obj.host/rewriter/gen_usage_rewriter_dictionary_main.gen_usage_rewriter_dictionary_main.o obj.host/base/libbase.a obj.host/base/libserialized_string_array.a obj.host/base/libbase_core.a obj.host/base/libclock.a obj.host/base/libsingleton.a obj.host/base/libmutex.a obj.host/base/libflags.a obj.host/base/libhash.a -Wl,--end-group  -labsl_base -labsl_int128 -labsl_strings -labsl_hash -labsl_city -labsl_strings_internal -labsl_flags_reflection -labsl_raw_hash_set -labsl_str_format_internal -labsl_throw_delegate -labsl_time_zone -labsl_hashtablez_sampler -labsl_synchronization -labsl_time
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: /nix/store/j1j9ffjhl54g8k418k8zjpmjc67p7vfs-abseil-cpp-20200923.2/lib/libabsl_time.a(clock.cc.o): in function `absl::lts_2020_09_23::GetCurrentTimeNanos()':
(.text+0x5): undefined reference to `absl::lts_2020_09_23::base_internal::UnscaledCycleClock::Now()'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: /nix/store/j1j9ffjhl54g8k418k8zjpmjc67p7vfs-abseil-cpp-20200923.2/lib/libabsl_time.a(clock.cc.o): in function `absl::lts_2020_09_23::GetCurrentTimeNanosSlowPath()':
(.text.unlikely+0x3f): undefined reference to `absl::lts_2020_09_23::base_internal::SpinLock::SlowLock()'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: (.text.unlikely+0x5e): undefined reference to `absl::lts_2020_09_23::base_internal::UnscaledCycleClock::Now()'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: (.text.unlikely+0x9c): undefined reference to `absl::lts_2020_09_23::raw_logging_internal::RawLog(absl::lts_2020_09_23::LogSeverity, char const*, int, char const*, ...)'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: (.text.unlikely+0xab): undefined reference to `absl::lts_2020_09_23::base_internal::UnscaledCycleClock::Now()'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: (.text.unlikely+0x3a4): undefined reference to `absl::lts_2020_09_23::base_internal::SpinLock::SlowUnlock(unsigned int)'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: /nix/store/j1j9ffjhl54g8k418k8zjpmjc67p7vfs-abseil-cpp-20200923.2/lib/libabsl_time.a(duration.cc.o): in function `absl::lts_2020_09_23::time_internal::IDivDuration(bool, absl::lts_2020_09_23::Duration, absl::lts_2020_09_23::Duration, absl::lts_2020_09_23::Duration*)':
(.text+0x6a8): undefined reference to `absl::lts_2020_09_23::operator/(absl::lts_2020_09_23::uint128, absl::lts_2020_09_23::uint128)'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: (.text+0x9e6): undefined reference to `absl::lts_2020_09_23::operator/(absl::lts_2020_09_23::uint128, absl::lts_2020_09_23::uint128)'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: /nix/store/j1j9ffjhl54g8k418k8zjpmjc67p7vfs-abseil-cpp-20200923.2/lib/libabsl_time.a(duration.cc.o): in function `absl::lts_2020_09_23::Duration::operator*=(long)':
(.text+0xddb): undefined reference to `absl::lts_2020_09_23::kuint128max'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: (.text+0xdf1): undefined reference to `absl::lts_2020_09_23::operator/(absl::lts_2020_09_23::uint128, absl::lts_2020_09_23::uint128)'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: (.text+0xe83): undefined reference to `absl::lts_2020_09_23::operator/(absl::lts_2020_09_23::uint128, absl::lts_2020_09_23::uint128)'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: /nix/store/j1j9ffjhl54g8k418k8zjpmjc67p7vfs-abseil-cpp-20200923.2/lib/libabsl_time.a(duration.cc.o): in function `absl::lts_2020_09_23::Duration::operator/=(long)':
(.text+0x1262): undefined reference to `absl::lts_2020_09_23::operator/(absl::lts_2020_09_23::uint128, absl::lts_2020_09_23::uint128)'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: (.text+0x134b): undefined reference to `absl::lts_2020_09_23::operator/(absl::lts_2020_09_23::uint128, absl::lts_2020_09_23::uint128)'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: obj.host/base/base_core.logging.o: in function `mozc::Logging::GetLogMessageHeader[abi:cxx11]()':
logging.cc:(.text+0x4a): undefined reference to `absl::lts_2020_09_23::FormatTime[abi:cxx11](std::basic_string_view<char, std::char_traits<char> >, absl::lts_2020_09_23::Time, absl::lts_2020_09_23::TimeZone)'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: obj.host/base/base_core.util.o: in function `mozc::Util::SplitStringUsing(std::basic_string_view<char, std::char_traits<char> >, char const*, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >*)':
util.cc:(.text+0x617): undefined reference to `absl::lts_2020_09_23::ByAnyChar::ByAnyChar(std::basic_string_view<char, std::char_traits<char> >)'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: obj.host/base/base_core.util.o: in function `mozc::Util::SplitStringUsing(std::basic_string_view<char, std::char_traits<char> >, char const*, std::vector<std::basic_string_view<char, std::char_traits<char> >, std::allocator<std::basic_string_view<char, std::char_traits<char> > > >*)':
util.cc:(.text+0x863): undefined reference to `absl::lts_2020_09_23::ByAnyChar::ByAnyChar(std::basic_string_view<char, std::char_traits<char> >)'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: obj.host/base/base_core.util.o: in function `mozc::Util::SplitStringAllowEmpty(std::basic_string_view<char, std::char_traits<char> >, char const*, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >*)':
util.cc:(.text+0x9f7): undefined reference to `absl::lts_2020_09_23::ByAnyChar::ByAnyChar(std::basic_string_view<char, std::char_traits<char> >)'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: obj.host/base/base_core.util.o: in function `mozc::Util::Unescape(std::basic_string_view<char, std::char_traits<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)':
util.cc:(.text+0x4bc3): undefined reference to `absl::lts_2020_09_23::CUnescape(std::basic_string_view<char, std::char_traits<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: obj.host/base/base_core.util.o: in function `absl::lts_2020_09_23::strings_internal::Splitter<absl::lts_2020_09_23::ByAnyChar, absl::lts_2020_09_23::SkipEmpty>::ConvertToContainer<std::vector<std::basic_string_view<char, std::char_traits<char> >, std::allocator<std::basic_string_view<char, std::char_traits<char> > > >, std::basic_string_view<char, std::char_traits<char> >, false>::operator()(absl::lts_2020_09_23::strings_internal::Splitter<absl::lts_2020_09_23::ByAnyChar, absl::lts_2020_09_23::SkipEmpty> const&) const':
util.cc:(.text._ZNK4absl14lts_2020_09_2316strings_internal8SplitterINS0_9ByAnyCharENS0_9SkipEmptyEE18ConvertToContainerISt6vectorISt17basic_string_viewIcSt11char_traitsIcEESaISB_EESB_Lb0EEclERKS5_[_ZNK4absl14lts_2020_09_2316strings_internal8SplitterINS0_9ByAnyCharENS0_9SkipEmptyEE18ConvertToContainerISt6vectorISt17basic_string_viewIcSt11char_traitsIcEESaISB_EESB_Lb0EEclERKS5_]+0x9a): undefined reference to `absl::lts_2020_09_23::ByAnyChar::Find(std::basic_string_view<char, std::char_traits<char> >, unsigned long) const'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: obj.host/base/base_core.util.o: in function `absl::lts_2020_09_23::strings_internal::SplitIterator<absl::lts_2020_09_23::strings_internal::Splitter<absl::lts_2020_09_23::ByAnyChar, absl::lts_2020_09_23::SkipEmpty> >::SplitIterator(absl::lts_2020_09_23::strings_internal::SplitIterator<absl::lts_2020_09_23::strings_internal::Splitter<absl::lts_2020_09_23::ByAnyChar, absl::lts_2020_09_23::SkipEmpty> >::State, absl::lts_2020_09_23::strings_internal::Splitter<absl::lts_2020_09_23::ByAnyChar, absl::lts_2020_09_23::SkipEmpty> const*)':
util.cc:(.text._ZN4absl14lts_2020_09_2316strings_internal13SplitIteratorINS1_8SplitterINS0_9ByAnyCharENS0_9SkipEmptyEEEEC2ENS7_5StateEPKS6_[_ZN4absl14lts_2020_09_2316strings_internal13SplitIteratorINS1_8SplitterINS0_9ByAnyCharENS0_9SkipEmptyEEEEC2ENS7_5StateEPKS6_]+0xf9): undefined reference to `absl::lts_2020_09_23::ByAnyChar::Find(std::basic_string_view<char, std::char_traits<char> >, unsigned long) const'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: obj.host/base/base_core.util.o: in function `absl::lts_2020_09_23::strings_internal::Splitter<absl::lts_2020_09_23::ByAnyChar, absl::lts_2020_09_23::AllowEmpty>::ConvertToContainer<std::vector<std::basic_string_view<char, std::char_traits<char> >, std::allocator<std::basic_string_view<char, std::char_traits<char> > > >, std::basic_string_view<char, std::char_traits<char> >, false>::operator()(absl::lts_2020_09_23::strings_internal::Splitter<absl::lts_2020_09_23::ByAnyChar, absl::lts_2020_09_23::AllowEmpty> const&) const':
util.cc:(.text._ZNK4absl14lts_2020_09_2316strings_internal8SplitterINS0_9ByAnyCharENS0_10AllowEmptyEE18ConvertToContainerISt6vectorISt17basic_string_viewIcSt11char_traitsIcEESaISB_EESB_Lb0EEclERKS5_[_ZNK4absl14lts_2020_09_2316strings_internal8SplitterINS0_9ByAnyCharENS0_10AllowEmptyEE18ConvertToContainerISt6vectorISt17basic_string_viewIcSt11char_traitsIcEESaISB_EESB_Lb0EEclERKS5_]+0xaa): undefined reference to `absl::lts_2020_09_23::ByAnyChar::Find(std::basic_string_view<char, std::char_traits<char> >, unsigned long) const'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: obj.host/base/base_core.util.o: in function `absl::lts_2020_09_23::strings_internal::SplitIterator<absl::lts_2020_09_23::strings_internal::Splitter<absl::lts_2020_09_23::ByAnyChar, absl::lts_2020_09_23::AllowEmpty> >::SplitIterator(absl::lts_2020_09_23::strings_internal::SplitIterator<absl::lts_2020_09_23::strings_internal::Splitter<absl::lts_2020_09_23::ByAnyChar, absl::lts_2020_09_23::AllowEmpty> >::State, absl::lts_2020_09_23::strings_internal::Splitter<absl::lts_2020_09_23::ByAnyChar, absl::lts_2020_09_23::AllowEmpty> const*)':
util.cc:(.text._ZN4absl14lts_2020_09_2316strings_internal13SplitIteratorINS1_8SplitterINS0_9ByAnyCharENS0_10AllowEmptyEEEEC2ENS7_5StateEPKS6_[_ZN4absl14lts_2020_09_2316strings_internal13SplitIteratorINS1_8SplitterINS0_9ByAnyCharENS0_10AllowEmptyEEEEC2ENS7_5StateEPKS6_]+0xe7): undefined reference to `absl::lts_2020_09_23::ByAnyChar::Find(std::basic_string_view<char, std::char_traits<char> >, unsigned long) const'
/nix/store/wcz8lsibkw2nikdb0bjl22p75lfwahzr-binutils-2.35.1/bin/ld: /nix/store/j1j9ffjhl54g8k418k8zjpmjc67p7vfs-abseil-cpp-20200923.2/lib/libabsl_strings.a(numbers.cc.o): in function `absl::lts_2020_09_23::SimpleAtob(absl::lts_2020_09_23::string_view, bool*)':
(.text+0x949): undefined reference to `absl::lts_2020_09_23::raw_logging_internal::RawLog(absl::lts_2020_09_23::LogSeverity, char const*, int, char const*, ...)'
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
[390/524] CXX obj.host/converter/gen_segmenter_bitarray.gen_segmenter_bitarray.oer_data.o[K
ninja: build stopped: subcommand failed.
Traceback (most recent call last):
  File "build_mozc.py", line 876, in <module>
    main()
  File "build_mozc.py", line 863, in main
    BuildMain(cmd_opts, cmd_args)
  File "build_mozc.py", line 608, in BuildMain
    BuildWithNinja(options, targets)
  File "build_mozc.py", line 582, in BuildWithNinja
    RunOrDie([ninja, '-C', build_arg, target_name])
  File "/build/source/src/build_tools/util.py", line 98, in RunOrDie
    raise RunOrDieError('\n'.join(['',
build_tools.util.RunOrDieError: 
==========
 ERROR: ninja -C out_linux/Release mozc_server
==========
error: --- Error -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- nix-build
builder for '/nix/store/m97k34xv7cfmsp09icx7jcdv158a0xrw-fcitx5-mozc-2.26.4220.102.drv' failed with exit code 1; last 10 log lines:
    File "build_mozc.py", line 608, in BuildMain
      BuildWithNinja(options, targets)
    File "build_mozc.py", line 582, in BuildWithNinja
      RunOrDie([ninja, '-C', build_arg, target_name])
    File "/build/source/src/build_tools/util.py", line 98, in RunOrDie
      raise RunOrDieError('\n'.join(['',
  build_tools.util.RunOrDieError: 
  ==========
   ERROR: ninja -C out_linux/Release mozc_server
  ==========
```

--- a/maintainers/maintainer-list.nix
+++ b/maintainers/maintainer-list.nix
@@ -1084,6 +1084,12 @@
     githubId = 75972;
     name = "Ben Booth";
   };
+  berberman = {
+    email = "berberman@yandex.com";
+    github = "berberman";
+    githubId = 26041945;
+    name = "Potato Hatsue";
+  };
   berce = {
     email = "bert.moens@gmail.com";
     github = "berce";

--- a/pkgs/development/misc/breakpad/default.nix
+++ b/pkgs/development/misc/breakpad/default.nix
@@ -1,0 +1,30 @@
+{ lib, stdenv, fetchgit }:
+let
+  lss = fetchgit {
+    url = "https://chromium.googlesource.com/linux-syscall-support";
+    rev = "d9ad2969b369a9f1c455fef92d04c7628f7f9eb8";
+    sha256 = "952dv+ZE1ge/WF5RyHmEqht+AofoRHKAeFmGasVF9BA=";
+  };
+in stdenv.mkDerivation {
+  pname = "breakpad";
+
+  version = "unstable-3b3469e";
+
+  src = fetchgit {
+    url = "https://chromium.googlesource.com/breakpad/breakpad";
+    rev = "3b3469e9ed0de3d02e4450b9b95014a4266cf2ff";
+    sha256 = "bRGOBrGPK+Zxp+KK+E5MFkYlDUNVhVeInVSwq+eCAF0=";
+  };
+
+  postUnpack = ''
+    ln -s ${lss} $sourceRoot/src/third_party/lss
+  '';
+
+  meta = with lib; {
+    description = "An open-source multi-platform crash reporting system";
+    homepage = "https://chromium.googlesource.com/breakpad";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ berberman ];
+    platforms = with platforms; linux ++ darwin ++ windows;
+  };
+}

--- a/pkgs/development/misc/breakpad/default.nix
+++ b/pkgs/development/misc/breakpad/default.nix
@@ -25,6 +25,6 @@ in stdenv.mkDerivation {
     homepage = "https://chromium.googlesource.com/breakpad";
     license = licenses.bsd3;
     maintainers = with maintainers; [ berberman ];
-    platforms = with platforms; linux ++ darwin ++ windows;
+    platforms = platforms.all;
   };
 }

--- a/pkgs/tools/inputmethods/fcitx5/fcitx5-mozc.nix
+++ b/pkgs/tools/inputmethods/fcitx5/fcitx5-mozc.nix
@@ -1,0 +1,117 @@
+{ lib, clangStdenv, fetchFromGitHub, fetchurl, fetchpatch, fetchgit
+, python3Packages, mesa, ninja, pkg-config, protobuf, zinnia, qt5, fcitx5
+, jsoncpp, gtest, which, gtk2, unzip, abseil-cpp, breakpad }:
+let
+  inherit (python3Packages) python gyp six;
+  japanese_usage_dictionary = fetchFromGitHub {
+    owner = "hiroyuki-komatsu";
+    repo = "japanese-usage-dictionary";
+    rev = "e5b3425575734c323e1d947009dd74709437b684";
+    sha256 = "0pyrpz9c8nxccwpgyr36w314mi8h132cis8ijvlqmmhqxwsi30hm";
+  };
+  # abseil-cpp in nixpkgs is too old
+  abseil-cpp_2020923 = abseil-cpp.overrideAttrs (old: rec {
+    version = "20200923.2";
+    src = fetchFromGitHub {
+      owner = "abseil";
+      repo = "abseil-cpp";
+      rev = version;
+      sha256 = "G+wkaC4IPtyc/xCUyVFJOcHppPFU7KkhIHjv6uhVKGU=";
+    };
+    cmakeFlags = [ "-DCMAKE_CXX_STANDARD=17" "-DBUILD_SHARED_LIBS=ON" ];
+  });
+  zipcode_rel = "202011";
+  jigyosyo = fetchurl {
+    url = "https://osdn.net/projects/ponsfoot-aur/storage/mozc/jigyosyo-${zipcode_rel}.zip";
+    sha256 = "j7MkNtd4+QTi91EreVig4/OV0o5y1+KIjEJBEmLK/mY=";
+  };
+  x-ken-all = fetchurl {
+    url =
+      "https://osdn.net/projects/ponsfoot-aur/storage/mozc/x-ken-all-${zipcode_rel}.zip";
+    sha256 = "ExS0Cg3rs0I9IOVbZHLt8UEfk8/LmY9oAHPVVlYuTPw=";
+  };
+
+in clangStdenv.mkDerivation rec {
+  pname = "fcitx5-mozc";
+  version = "2.26.4220.102";
+
+  src = fetchFromGitHub {
+    owner = "fcitx";
+    repo = "mozc";
+    rev = "1882e33b61673b66d63277f82b4c80ae4e506c10";
+    sha256 = "R+w0slVFpqtt7PIr1pyupJjRoQsABVZiMdZ9fKGKAqw=";
+  };
+
+  nativeBuildInputs = [ gyp ninja mesa python pkg-config qt5.wrapQtAppsHook six which unzip ];
+
+  buildInputs = [ protobuf zinnia qt5.qtbase fcitx5 abseil-cpp_2020923 jsoncpp gtest gtk2 ];
+
+  patches = [
+    # Support linking system abseil-cpp
+    (fetchpatch {
+      url = "https://salsa.debian.org/debian/mozc/-/raw/debian/sid/debian/patches/0007-Update-src-base-absl.gyp.patch";
+      sha256 = "UiS0UScDKyAusXOhc7Bg8dF8ARQQiVTylEhAOxqaZt8=";
+    })
+
+  ];
+
+  postUnpack = ''
+    unzip ${x-ken-all} -d $sourceRoot/src/
+    unzip ${jigyosyo} -d $sourceRoot/src/
+
+    rmdir $sourceRoot/src/third_party/breakpad/
+    ln -s ${breakpad} $sourceRoot/src/third_party/breakpad
+    rmdir $sourceRoot/src/third_party/gtest/
+    ln -s ${gtest} $sourceRoot/src/third_party/gtest
+    rmdir $sourceRoot/src/third_party/gyp/
+    ln -s ${gyp} $sourceRoot/src/third_party/gyp
+    rmdir $sourceRoot/src/third_party/japanese_usage_dictionary/
+    ln -s ${japanese_usage_dictionary} $sourceRoot/src/third_party/japanese_usage_dictionary
+  '';
+
+  # Copied from https://github.com/archlinux/svntogit-community/blob/packages/fcitx5-mozc/trunk/PKGBUILD
+  configurePhase = ''
+    cd src
+    export GYP_DEFINES="document_dir=$out/share/doc/mozc use_libzinnia=1 use_libprotobuf=1 use_libabseil=1"
+
+    # disable fcitx4
+    rm unix/fcitx/fcitx.gyp
+
+    # gen zip code seed
+    PYTHONPATH="$PWD:$PYTHONPATH" python dictionary/gen_zip_code_seed.py --zip_code="x-ken-all.csv" --jigyosyo="JIGYOSYO.CSV" >> data/dictionary_oss/dictionary09.txt
+
+    # use libstdc++ instead of libc++
+    sed "/stdlib=libc++/d;/-lc++/d" -i gyp/common.gypi
+
+    # run gyp
+    python build_mozc.py gyp --gypdir=${gyp}/bin --server_dir=$out/lib/mozc
+  '';
+
+  buildPhase = ''
+    python build_mozc.py build -c Release \
+      server/server.gyp:mozc_server \
+      gui/gui.gyp:mozc_tool \
+      unix/fcitx5/fcitx5.gyp:fcitx5-mozc
+  '';
+
+  installPhase = ''
+    export PREFIX=$out
+    export _bldtype=Release
+    ../scripts/install_server
+    install -d $out/share/licenses/fcitx5-mozc
+    head -n 29 server/mozc_server.cc > $out/share/licenses/fcitx5-mozc/LICENSE
+    install -m644 data/installer/*.html $out/share/licenses/fcitx5-mozc/
+    install -d $out/share/fcitx5/addon
+    install -d $out/share/fcitx5/inputmethod
+    install -d $out/lib/fcitx5
+    ../scripts/install_fcitx5
+  '';
+
+  meta = with lib; {
+    description = "Fcitx5 Module of A Japanese Input Method for Chromium OS, Windows, Mac and Linux (the Open Source Edition of Google Japanese Input)";
+    homepage = "https://github.com/fcitx/mozc";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ berberman ];
+    platforms = platforms.linux;
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -156,6 +156,8 @@ in
 
   bacnet-stack = callPackage ../tools/networking/bacnet-stack {};
 
+  breakpad = callPackage ../development/misc/breakpad {};
+
   # Zip file format only allows times after year 1980, which makes e.g. Python wheel building fail with:
   # ValueError: ZIP does not support timestamps before 1980
   ensureNewerSourcesForZipFilesHook = ensureNewerSourcesHook { year = "1980"; };

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -156,7 +156,7 @@ in
 
   bacnet-stack = callPackage ../tools/networking/bacnet-stack {};
 
-  breakpad = callPackage ../development/misc/breakpad {};
+  breakpad = callPackage ../development/misc/breakpad { };
 
   # Zip file format only allows times after year 1980, which makes e.g. Python wheel building fail with:
   # ValueError: ZIP does not support timestamps before 1980
@@ -4016,6 +4016,8 @@ in
   fcitx5-with-addons = libsForQt5.callPackage ../tools/inputmethods/fcitx5/with-addons.nix { };
 
   fcitx5-chinese-addons = libsForQt5.callPackage ../tools/inputmethods/fcitx5/fcitx5-chinese-addons.nix { };
+
+  fcitx5-mozc = libsForQt5.callPackage ../tools/inputmethods/fcitx5/fcitx5-mozc.nix { };
 
   fcitx5-configtool = libsForQt5.callPackage ../tools/inputmethods/fcitx5/fcitx5-configtool.nix { };
 

